src/main/scala/org/globalforestwatch/summarystats/gfwpro_dashboard/GfwProDashboardSummary.scala

```diff
@@ -26,11 +26,23 @@ case class GfwProDashboardSummary(
   def isEmpty = stats.isEmpty
 
   /** Pivot raw data to GfwProDashboardData and aggregate across alert dates. */
-  def toGfwProDashboardData(): GfwProDashboardData = {
-    stats
-      .map { case (group, data) => group.
-        toGfwProDashboardData(data.alertCount, data.treeCoverExtentArea) }
-      .foldLeft(GfwProDashboardData.empty)( _ merge _)
+  def toGfwProDashboardData(ignoreGadm: Boolean): List[GfwProDashboardData] = {
+    if (ignoreGadm) {
```

Comment:
This works, but I feel like having the split between raster for small number of features and vector for big number of features adds a lot of complication/splitting paths to the code. 

Is there a good way to do raster for all cases eventually? I think mapping polygons to centroids from an RDD is pretty fast, you could potentially pull those all out into a separate DF, repartition those points by spatial keys, and then get the Gadm ID for each. But I guess that's also fairly complicated.

Reply:
Yes, I think your suggested way would add some  complexity too.  But the bigger problem is that I found that the vector GADM method is more efficient for very large dataset (100s of locations).  I think it's just that with 100s of locations, the vector intersection method eventually becomes more efficient than the raster method.  This was even true when I tried some extra complexities to compute the GADMs of the centroids of the locations during the GridVisitor (so tiles with centroids were not loaded a second time).

I do realize this optimization adds complexity, but I'm really hoping we get a big win for all the hundreds of small jobs on Saturday night, which can run in less than a minute with the raster GADM method, but take 5-8 minutes at minimum for the vector GADM method, because they have to load the entire vector GADM file.